cirq/devices/line_qubit.py

```diff
@@ -40,6 +40,19 @@ def is_adjacent(self, other: ops.Qid) -> bool:
         """Determines if two qubits are adjacent line qubits."""
         return isinstance(other, _BaseLineQid) and abs(self.x - other.x) == 1
 
+    def neighbors(self, qids: Optional[Iterable[ops.Qid]] = None
+                 ) -> Set['_BaseLineQid']:
```

Comment:
This is fine for now, but could use a self-type to make this more precise. Something like:
```python
T = TypeVar('T', bound='_BaseLineQid', covariant=True)

class _BaseLineQid(ops.Qid):
    ...
    def neighbors(self: T, ...) -> Set[T]:
        ...
```